packages/eslint-plugin/docs/rules/return-await.mdx

```diff
@@ -214,6 +218,35 @@ async function validAlways3() {
 </TabItem>
 </Tabs>
 
+### `error-handling-correctness-only`
```

Comment:
[Refactor] I think the table being useful is a symptom of the options being kind of unclear. What do you think about switching this to be a `boolean` option people have to enable with an options object?

```js
"@typescript-eslint/return-await": [
  "error",
  "always",
  { errorHandlingCorrectnessOnly: true }
]
```

Since we'd rather folks use `'always'` or `'in-try-catch'` anyway, I don't mind the mild user pain of an extra object. And I think it'd make the implementation simpler too!

Reply:
Documenting private conversation - Leaning against this, on the assumption for now that `never` is on its way out (a la https://github.com/typescript-eslint/typescript-eslint/pull/9364#discussion_r1651963801), in which case the flat options of (always|in-try-catch|error-handling-correctness-only) do make the most sense